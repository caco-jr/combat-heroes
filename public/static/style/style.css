:root,
[data-theme='default'] {
  /* // --- Base Colors */
  --light-grey: #f4f4f4;
  --grey: #bababa;
  --dark-grey: #7c7c84;
  --blue: #1ea2e0;
  --darker-blue: #0a0818;
  --green: #0ad0b5;
  --pink: #d83f87;
  --red: #d21437;
  --dark-red: #d93232;
  --purple: #2a2356;
  --dark-purple: #201a41;
  --yellow: #d7b529;
  /* // ------------------------- */
  /* // --- Brand colors */
  --primary-color: var(--red);
  --secondary-color: var(--green);
  /* // ------------------------- */
  /* // --- Status color */
  --error-color: var(--pink);
  /* // ------------------------- */
  /* // --- Border */
  --border-color: var(--grey);
  --border-radius: 10px;
  /* // ------------------------- */
  /* // --- Others */
  --background-color: #1d1e26;
  --text-color: #fff;
  --title-color: #fff;
  /* // ------------------------- */
  /* // --- Font */
  --primary-font-family: 'DM Sans', Helvetica, sans-serif;
  --text-font-family: var(--primary-font-family);
  --text-font-weight: 400;
  --title-font-family: var(--primary-font-family);
  --title-font-weight: 700;
  /* // ------------------------- */
}

[data-theme] {
  background-color: var(--background-color);
  color: var(--text-color);
}

[data-theme='light'] {
  --background-color: #fff;
  --text-color: #111;
  --title-color: #111;
}

* {
  font-family: var(--primary-font-family);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

img {
  border-radius: var(--border-radius);
  height: auto;
  max-width: 100%;
}

.container {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

@media (min-width: 768px) {
  .container {
    width: 750px;
  }
}

@media (min-width: 992px) {
  .container {
    width: 970px;
  }
}

@media (min-width: 1200px) {
  .container {
    width: 1170px;
  }
}
