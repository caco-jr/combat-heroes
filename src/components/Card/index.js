import React from 'react';

const Card = ({ name, imageURL, id }) => {
  return (
    <section>
      <img src={imageURL} />
      <span>{name} </span>
    </section>
  );
};

export default Card;
