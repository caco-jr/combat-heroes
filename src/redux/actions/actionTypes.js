export const ADD_TO_SLOT = 'ADD_TO_SLOT';
export const REMOVE_FROM_SLOT = 'REMOVE_FROM_SLOT';
