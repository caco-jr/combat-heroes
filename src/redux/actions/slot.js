import { ADD_TO_SLOT, REMOVE_FROM_SLOT } from './actionTypes';
import { getCharacterAPI } from '../../services/api';

export const addToSlot = (who, object) => dispatch => {
  const fields = `{
    id
    name
    image {
      url
    }
    biography {
      fullName
      placeOfBirth
    }
  }`;

  getCharacterAPI(object.id, fields).then(res => {
    dispatch({ type: ADD_TO_SLOT, who, object: res.data.character });
  });
};

export const removeFromSlot = who => dispatch =>
  dispatch({ type: REMOVE_FROM_SLOT, who });
