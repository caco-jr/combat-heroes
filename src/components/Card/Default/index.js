import React from 'react';
import { connect } from 'react-redux';

import { addToSlot } from '../../../redux/actions/slot';

const Card = ({ name, imageURL, id, slot, addToSlot }) => {
  return (
    <section>
      <img src={imageURL} />
      <button
        type="button"
        onClick={() => addToSlot('playerOne', { id })}
        disabled={slot.playerTwo.id === id}
      >
        A
      </button>

      <button
        type="button"
        onClick={() => addToSlot('playerTwo', { id })}
        disabled={slot.playerOne.id === id}
      >
        B
      </button>

      <p>{name} </p>
    </section>
  );
};

function mapStateToProps({ slot }) {
  return { slot };
}

function mapDispatchToProps(dispatch) {
  return {
    addToSlot: (who, object) => dispatch(addToSlot(who, object))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Card);
