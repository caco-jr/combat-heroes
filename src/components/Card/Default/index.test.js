import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Provider } from 'react-redux';

import Card from './index';
import makeStore from '../../../redux';
import { ADD_TO_SLOT } from '../../../redux/actions/actionTypes';
import { CardTitle } from './index.style';

const mock = {
  id: 39,
  name: 'Batman',
  image: {
    url: ''
  }
};

describe('With Enzyme', () => {
  it('renders', () => {
    const wrapper = shallow(
      <Provider store={makeStore()}>
        <Card {...mock} />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should card render character name', () => {
    const wrapper = mount(
      <Provider store={makeStore()}>
        <Card {...mock} />
      </Provider>
    );

    expect(wrapper.find(CardTitle).text()).toEqual(mock.name);
  });

  //   it('should select current character to slot', () => {
  //     const characterMock = {
  //       id: '208',
  //       name: 'Darth Vader',
  //       image: {
  //         url: ''
  //       },
  //       biography: {
  //         fullName: 'Anakin Skywalker',
  //         placeOfBirth: 'Tatooine'
  //       },
  //       slotName: 'playerOne'
  //     };

  //     const wrapper = mount(
  //       <Provider store={makeStore()}>
  //         <Card {...mock} />
  //       </Provider>
  //     );

  //     // makeStore().dispatch({
  //     //   type: ADD_TO_SLOT,
  //     //   who: { id: characterMock.id },
  //     //   object: characterMock
  //     // });

  //     expect(wrapper.find(CardTitle).text()).toEqual(characterMock.name);
  //   });
});

describe('With Snapshot Testing', () => {
  it('Rendering Preview', () => {
    const component = renderer.create(
      <Provider store={makeStore()}>
        <Card />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
