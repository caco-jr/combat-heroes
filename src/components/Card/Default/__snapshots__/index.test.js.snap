// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Enzyme renders 1`] = `ShallowWrapper {}`;

exports[`With Snapshot Testing Rendering Preview 1`] = `
<section>
  <img />
  <button
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    playerOne
  </button>
  <button
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    playerTwo
  </button>
  <h3
    className=""
  />
</section>
`;
